uxarray/core/dataarray.py

```diff
@@ -375,11 +375,29 @@ def to_raster(
             _RasterAxAttrs,
         )
 
-        _ensure_dimensions(self)
+        data = _ensure_dimensions(self)
 
         if not isinstance(ax, GeoAxes):
             raise TypeError("`ax` must be an instance of cartopy.mpl.geoaxes.GeoAxes")
 
+        auto_extent_applied = False
+        if pixel_mapping is None:
+            xlim, ylim = ax.get_xlim(), ax.get_ylim()
+            if np.allclose(xlim, (0.0, 1.0)) and np.allclose(ylim, (0.0, 1.0)):
+                try:
+                    import cartopy.crs as ccrs
+
+                    lon = self.uxgrid.node_lon.values
+                    lat = self.uxgrid.node_lat.values
```

Comment:
Please avoid converting the whole node coords to numpy, i.e. only convert the final scalars after finding min -max via xarray objects